File: src/src/CodeInTasks.Web.Models/Solution/SolutionViewModel.cs
namespace CodeInTasks.Web.Models.Solution
{
    public class SolutionViewModel
    {
        public Guid Id { get; set; }
        public string RepositoryUrl { get; set; }

        public TaskSolutionStatus Status { get; set; }
        public TaskSolutionResult? Result { get; set; }
        public string ResultAdditionalInfo { get; set; }

        public DateTime SendTime { get; set; }
        public DateTime? FinishTime { get; set; }
        public int? RunTimeMs { get; set; }

        public Guid TaskId { get; set; }
        public Guid SenderId { get; set; }
    }
}

File: frontend/CodeInTasks/src/app/web-api/models/solution/solution-view-model.ts
import { TaskSolutionResult, TaskSolutionStatus } from "../../enums";


export class SolutionViewModel {
    public id: string = null!;
    public repositoryUrl: string = null!;

    public status: TaskSolutionStatus = null!;
    public result: TaskSolutionResult | null = null;
    public errorCode: string | null = null;
    public resultAdditionalInfo: string | null = null;

    public sendTime: Date = null!;
    public finishTime: Date | null = null;
    public runTimeMs: number | null = null;

    public taskId: string = null!;
    public senderId: string = null!;
}
